### ISSUE TITLE
Add client-side validations to all forms

### ISSUE THREAD

**Issue by scottyschup** [MidwestAccessCoalition/lockbox_rails#246]
## Context
Currently most forms don't have any sort of client-side validation. The following files have at least one `.form-control` element and may need validations added:
* `app/views/devise/passwords/edit.html.erb`
* `app/views/devise/passwords/new.html.erb`
* `app/views/devise/sessions/new.html.erb`
* `app/views/lockbox_partners/add_cash/new.html.erb`
* `app/views/lockbox_partners/reconciliation/new.html.erb`
* `app/views/lockbox_partners/support_requests/_lockbox_transaction_fields.html.erb`
* `app/views/lockbox_partners/support_requests/_lockbox_transaction_form.html.erb`
* `app/views/lockbox_partners/support_requests/edit.html.erb`
* `app/views/lockbox_partners/support_requests/new.html.erb`
* `app/views/lockbox_partners/users/new.html.erb`

## Acceptance Criteria
`app/views/lockbox_partners/_lockbox_partner_inputs.html.erb` already has basic HTML5 client-side validations and can be used as a pattern to follow. It utilizes JS from `app/javascript/src/form_validation.js` that sets and removes the CSS class `.pristine` to ensure that forms are not marked as invalid on page load, but only after the inputs have been interacted with or an attempt to submit the form has occurred.

## Resources
See related issue #228 and its corresponding PR #242

**Comment by BintLopez** [comment]
@scottyschup -- is there still work to do for this issue or was it resolved with 242? cc @jel535s

**Comment by scottyschup** [comment]
Yeah, #242 was just for lockbox partner forms. This is the follow-up issue for all other forms.

**Comment by scottyschup** [comment]
I was going to take this, but since I'll be out this week, if someone else wants to, feel free. Otherwise I'll pick it up next week.
